src/execution_plan/optimizations/seek_by_id.c

```diff
@@ -73,7 +75,7 @@ static void _UseIdOptimization(ExecutionPlan *plan, OpBase *scan_op) {
 			OpBase_Free((OpBase *)filter);
```

Comment:
Maybe we should delay this free, accumulate all converted ops and free them at the very end.

Reply:
I don't really feel like that would be worth it - we don't know the max number of ops to initialize a stack array with, and a flexible array would be unnecessary allocations and complexity.